commit ebca61c40c4aa6536f63f68b01ff127991f824cf
Author: Isaías A. Comprés Ureña <isaias.compres@tum.de>
Date:   Wed Mar 9 15:01:11 2022 +0100

    Enabled Open PMIx up to version 5.

diff --git a/auxdir/x_ac_pmix.m4 b/auxdir/x_ac_pmix.m4
index 0fabec463c..b9e652e305 100755
--- a/auxdir/x_ac_pmix.m4
+++ b/auxdir/x_ac_pmix.m4
@@ -2,6 +2,7 @@
 #  AUTHOR:
 #    Artem Polyakov <artpol84@gmail.com>
 #    Ralph Castain <ralph.h.castain@intel.com>
+#    Isaias Compres <isaias.compres@tum.de>
 #
 #  SYNOPSIS:
 #    X_AC_PMIX
@@ -18,6 +19,7 @@ AC_DEFUN([X_AC_PMIX],
   _x_ac_pmix_v2_found="0"
   _x_ac_pmix_v3_found="0"
   _x_ac_pmix_v4_found="0"
+  _x_ac_pmix_v5_found="0"
 
   AC_ARG_WITH(
     [pmix],
@@ -59,6 +61,13 @@ AC_DEFUN([X_AC_PMIX],
 
             _x_ac_pmix_version="0"
             AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
+              #include <pmix_version.h>
+              #if (PMIX_VERSION_MAJOR != 5L)
+                #error "not version 5"
+              #endif
+            ], [ ] )],
+            [ _x_ac_pmix_version="5" ],
+            [ AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
               #include <pmix_version.h>
               #if (PMIX_VERSION_MAJOR != 4L)
                 #error "not version 4"
@@ -81,6 +90,7 @@ AC_DEFUN([X_AC_PMIX],
             [ _x_ac_pmix_version="2" ] )
             ])
             ])
+            ])
 
             CPPFLAGS="$_x_ac_pmix_cppflags_save"
             LIBS="$_x_ac_pmix_libs_save"
@@ -129,23 +139,43 @@ AC_DEFUN([X_AC_PMIX],
 	    # happen. If v4 is important to you please contact them instead of
 	    # opening a bug with SchedMD.
 
-            # if [test "$_x_ac_pmix_version" = "4"]; then
-            #   if [test "$_x_ac_pmix_v4_found" = "1" ]; then
-            #     m4_define([err_pmix_v4],[error processing $x_ac_cv_pmix_libdir: PMIx v4.x])
-            #     AC_MSG_ERROR(err_pmix_v4 err_pmix)
-            #   fi
-            #   _x_ac_pmix_v4_found="1"
-            #   PMIX_V4_CPPFLAGS="-I$x_ac_cv_pmix_dir/include"
-            #   if test "$ac_with_rpath" = "yes"; then
-            #     PMIX_V4_LDFLAGS="-Wl,-rpath -Wl,$x_ac_cv_pmix_libdir -L$x_ac_cv_pmix_libdir"
-            #   else
-            #     PMIX_V4_CPPFLAGS=$PMIX_V4_CPPFLAGS" -DPMIXP_V4_LIBPATH=\\\"$x_ac_cv_pmix_libdir\\\""
-            #   fi
-            #   # We don't want to search the other lib after we found it in
-            #   # one place or we might report a false duplicate if lib64 is a
-            #   # symlink of lib.
-            #   break
-            # fi
+            if [test "$_x_ac_pmix_version" = "4"]; then
+              if [test "$_x_ac_pmix_v4_found" = "1" ]; then
+                m4_define([err_pmix_v4],[error processing $x_ac_cv_pmix_libdir: PMIx v4.x])
+                AC_MSG_ERROR(err_pmix_v4 err_pmix)
+              fi
+              _x_ac_pmix_v4_found="1"
+              PMIX_V4_CPPFLAGS="-I$x_ac_cv_pmix_dir/include"
+              if test "$ac_with_rpath" = "yes"; then
+                PMIX_V4_LDFLAGS="-Wl,-rpath -Wl,$x_ac_cv_pmix_libdir -L$x_ac_cv_pmix_libdir"
+              else
+                PMIX_V4_CPPFLAGS=$PMIX_V4_CPPFLAGS" -DPMIXP_V4_LIBPATH=\\\"$x_ac_cv_pmix_libdir\\\""
+              fi
+              # We don't want to search the other lib after we found it in
+              # one place or we might report a false duplicate if lib64 is a
+              # symlink of lib.
+              break
+            fi
+
+            if [test "$_x_ac_pmix_version" = "5"]; then
+              if [test "$_x_ac_pmix_v5_found" = "1" ]; then
+                m4_define([err_pmix_v5],[error processing $x_ac_cv_pmix_libdir: PMIx v5.x])
+                AC_MSG_ERROR(err_pmix_v5 err_pmix)
+              fi
+              _x_ac_pmix_v5_found="1"
+              PMIX_V5_CPPFLAGS="-I$x_ac_cv_pmix_dir/include"
+              if test "$ac_with_rpath" = "yes"; then
+                PMIX_V5_LDFLAGS="-Wl,-rpath -Wl,$x_ac_cv_pmix_libdir -L$x_ac_cv_pmix_libdir"
+              else
+                PMIX_V5_CPPFLAGS=$PMIX_V5_CPPFLAGS" -DPMIXP_V5_LIBPATH=\\\"$x_ac_cv_pmix_libdir\\\""
+              fi
+              # We don't want to search the other lib after we found it in
+              # one place or we might report a false duplicate if lib64 is a
+              # symlink of lib.
+              break
+            fi
+
+
           done
         done
       ])
@@ -158,9 +188,11 @@ AC_DEFUN([X_AC_PMIX],
     AC_SUBST(PMIX_V3_LDFLAGS)
     AC_SUBST(PMIX_V4_CPPFLAGS)
     AC_SUBST(PMIX_V4_LDFLAGS)
+    AC_SUBST(PMIX_V5_CPPFLAGS)
+    AC_SUBST(PMIX_V5_LDFLAGS)
 
-    if test $_x_ac_pmix_v2_found = 0 &&
-          test $_x_ac_pmix_v3_found = 0 && test $_x_ac_pmix_v4_found = 0; then
+    if test $_x_ac_pmix_v2_found = 0 && test $_x_ac_pmix_v3_found = 0 && 
+          test $_x_ac_pmix_v4_found = 0 && test $_x_ac_pmix_v5_found = 0; then
       if test -z "$with_pmix"; then
         AC_MSG_WARN([unable to locate pmix installation])
       else
@@ -175,4 +207,5 @@ AC_DEFUN([X_AC_PMIX],
   AM_CONDITIONAL(HAVE_PMIX_V2, [test $_x_ac_pmix_v2_found = "1"])
   AM_CONDITIONAL(HAVE_PMIX_V3, [test $_x_ac_pmix_v3_found = "1"])
   AM_CONDITIONAL(HAVE_PMIX_V4, [test $_x_ac_pmix_v4_found = "1"])
+  AM_CONDITIONAL(HAVE_PMIX_V5, [test $_x_ac_pmix_v5_found = "1"])
 ])
diff --git a/src/plugins/mpi/pmix/Makefile.am b/src/plugins/mpi/pmix/Makefile.am
index 57bf36d8b7..1f2d9af80f 100755
--- a/src/plugins/mpi/pmix/Makefile.am
+++ b/src/plugins/mpi/pmix/Makefile.am
@@ -63,6 +63,22 @@ mpi_pmix_v4_la_CPPFLAGS = $(AM_CPPFLAGS) $(PMIX_V4_CPPFLAGS) -DHAVE_PMIX_VER=4
 
 endif
 
+if HAVE_PMIX_V5
+
+pkglib_v5dir=$(pkglibdir)
+pkglib_v5_LTLIBRARIES = mpi_pmix_v5.la
+mpi_pmix_v5_la_SOURCES = $(pmix_src) pmixp_client_v2.c
+mpi_pmix_v5_la_LIBADD = $(pmix_libadd)
+mpi_pmix_v5_la_LDFLAGS = $(pmix_ldflags) $(PMIX_V4_LDFLAGS)
+mpi_pmix_v5_la_CPPFLAGS = $(AM_CPPFLAGS) $(PMIX_V4_CPPFLAGS) -DHAVE_PMIX_VER=5
+
+endif
+
+
+if HAVE_PMIX_V5
+mpi_pmix_so := mpi_pmix_v5.so
+else
+
 if HAVE_PMIX_V4
 mpi_pmix_so := mpi_pmix_v4.so
 else
@@ -77,6 +93,7 @@ mpi_pmix_so := mpi_pmix_v2.so
 endif
 endif
 endif
+endif
 
 install-exec-hook:
 	rm -rf $(DESTDIR)$(pkglibdir)/mpi_pmix.so
diff --git a/src/plugins/mpi/pmix/mpi_pmix.c b/src/plugins/mpi/pmix/mpi_pmix.c
index 79592f0cd3..bfd0b64e56 100644
--- a/src/plugins/mpi/pmix/mpi_pmix.c
+++ b/src/plugins/mpi/pmix/mpi_pmix.c
@@ -8,6 +8,8 @@
  *                          and Information Sciences (SibSUTIS).
  *                          All rights reserved.
  *  Written by Boris Bochkarev <boris-bochkaryov@yandex.ru>.
+ *  Copyright (C) 2022 Technical University of Munich. All rights reserved.
+ *  Written by Isaias Compres <isaias.compres@tum.de>.
  *
  *  This file is part of Slurm, a resource management program.
  *  For details, see <https://slurm.schedmd.com/>.
@@ -85,6 +87,10 @@ const char plugin_name[] = "PMIx plugin";
 const char plugin_type[] = "mpi/pmix_v2";
 #elif (HAVE_PMIX_VER == 3)
 const char plugin_type[] = "mpi/pmix_v3";
+#elif (HAVE_PMIX_VER == 4)
+const char plugin_type[] = "mpi/pmix_v4";
+#elif (HAVE_PMIX_VER == 5)
+const char plugin_type[] = "mpi/pmix_v5";
 #endif
 
 const uint32_t plugin_version = SLURM_VERSION_NUMBER;
@@ -110,6 +116,10 @@ static void *_libpmix_open(void)
 	xstrfmtcat(full_path, "%s/", PMIXP_V2_LIBPATH);
 #elif defined PMIXP_V3_LIBPATH
 	xstrfmtcat(full_path, "%s/", PMIXP_V3_LIBPATH);
+#elif defined PMIXP_V4_LIBPATH
+	xstrfmtcat(full_path, "%s/", PMIXP_V4_LIBPATH);
+#elif defined PMIXP_V5_LIBPATH
+	xstrfmtcat(full_path, "%s/", PMIXP_V5_LIBPATH);
 #endif
 	xstrfmtcat(full_path, "libpmix.so");
 
diff --git a/src/plugins/mpi/pmix/pmixp_client_v2.c b/src/plugins/mpi/pmix/pmixp_client_v2.c
index 7a1c0b66cb..f71aa1d1ab 100644
--- a/src/plugins/mpi/pmix/pmixp_client_v2.c
+++ b/src/plugins/mpi/pmix/pmixp_client_v2.c
@@ -9,6 +9,8 @@
  *                          and Information Sciences (SibSUTIS).
  *                          All rights reserved.
  *  Written by Boris Bochkarev <boris-bochkaryov@yandex.ru>.
+ *  Copyright (C) 2022 Technical University of Munich. All rights reserved.
+ *  Written by Isaias Compres <isaias.compres@tum.de>.
  *
  *  This file is part of Slurm, a resource management program.
  *  For details, see <https://slurm.schedmd.com/>.
@@ -52,6 +54,7 @@
 
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <pmix.h>
 #include <pmix_server.h>
 
 static int _client_connected(const pmix_proc_t *proc, void *server_object,
